src/ExpressReceiver.spec.ts

```diff
@@ -130,6 +141,34 @@ describe('ExpressReceiver', () => {
       await verifyMissingHeaderDetection(untypedReq as Request);
     });
 
+    // ----------------------------
+    // verifyInvalidTimestampError
+
+    function verifyInvalidTimestampError(req: Request): Promise<any> {
+      // Arrange
+      const resp = {} as Response;
+      let errorResult: any;
+      const next = (error: any) => { errorResult = error; };
+
+      // Act
+      const verifier = verifySignatureAndParseBody(noopLogger, signingSecret);
+      return verifier(req, resp, next).then((_: any) => {
+        // Assert
+        assert.equal(errorResult, 'Error: Slack request signing verification failed. Timestamp is invalid.');
+      });
+    }
+
+    it('should detect invalid timestamp header', async () => {
+      const reqAsStream = new Readable();
+      reqAsStream.push(body);
+      reqAsStream.push(null); // indicate EOF
+      (reqAsStream as { [key: string]: any }).headers = {
+        'x-slack-signature': signature,
+        'x-slack-request-timestamp': 'Hello there!',
+      };
+      await verifyInvalidTimestampError(reqAsStream as Request);
+    });
```

Comment:
this tests that if the timestamp isn't a real timestamp they get a useful error. this fell out of the `NaN` discussion, since you may either 
1. Not get a timestamp (undefined)
2. Get an invalid timstamp (can't convert to a number, so end up getting NaN)
3. Get a valid but old timestamp
4. Get a valid timestamp.